ex2_services/Services/source/general.c

```diff
@@ -80,26 +80,28 @@ void general_service(void *param) {
 
     for (;;) {
         svc_wdt_counter++;
-
+        csp_conn_t *conn;
         csp_packet_t *packet;
+
         // wait for connection, timeout
         if ((conn = csp_accept(sock, DELAY_WAIT_TIMEOUT)) == NULL) {
             svc_wdt_counter++;
             /* timeout */
             continue;
         }
         svc_wdt_counter++;
-        while ((packet = csp_read(conn, CSP_TIMEOUT)) != NULL) {
-            if (general_app(packet) != SATR_OK) {
-                // something went wrong, this shouldn't happen
+
+        while ((packet = csp_read(conn, 50)) != NULL) {
+            if (general_app(conn, packet) != SATR_OK) {
                 csp_buffer_free(packet);
-            } else {
+                ex2_log("Error responding to packet");
+            } else{
```

Comment:
Did you clang-format this? I think it would have caught this

Reply:
Haha you're right of course